File: 553. Optimal Division/Java/Optimal Division.java
public class Solution {
    public String optimalDivision(int[] nums) {
        int length = nums.length;
        StringBuilder builder = new StringBuilder();
        if (length > 0) {
            builder.append(nums[0]);
        }
        if (length > 1) {
            builder.append('/');
        }
        if (length > 2) {
            builder.append('(');
        }
        for (int i = 1; i < length - 1; i++) {
            builder.append(nums[i]);
            builder.append('/');
        }
        if (length > 1) {
            builder.append(nums[length - 1]);
        }
        if (length > 2) {
            builder.append(')');
        }
        return builder.toString();
    }
}

File: 553. Optimal Division/C++/Optimal Division.cpp
class Solution {
public:
    string optimalDivision(vector<int> &nums) {
        int length = nums.size();
        string result = "";
        if (length > 0) {
            result += to_string(nums[0]);
        }
        if (length > 1) {
            result += '/';
        }
        if (length > 2) {
            result += '(';
        }
        for (int i = 1; i < length - 1; i++) {
            result += to_string(nums[i]);
            result += '/';
        }
        if (length > 1) {
            result += to_string(nums[length - 1]);
        }
        if (length > 2) {
            result += ')';
        }
        return result;
    }
};